folly/Synchronized.h

```diff
@@ -135,6 +135,7 @@ class SynchronizedBase<Subclass, true> {
    *     return data.getValue();
    *   });
    */
+#if __cplusplus > 201103L
```

Comment:
After seeing your comment, I gave this a shot, but wasn't successful.  The best I could come up with is below, but in the meantime I did find this commit https://github.com/facebook/folly/commit/9fc18f0ac3b21dc2abb5700fba9fc0c27ffd190d#diff-0aeb0bf602d25a066eb2233e06c4a981R48 in which you move away from C++11.  

My PR is based on the condition that C++11 is supported, but if it's not (and that's totally reasonable) then I'm happy to withdraw the PR

-> typename std::result_of<Function(Subclass::DataType&)>::type